doltpy/core/write/write.py

```diff
@@ -85,3 +86,165 @@ def _import_helper(repo: Dolt,
+        tuple_list.append(tuple([data[col][i] for col in sorted_cols]))
+
+    insert_statement = _get_insert_statement(table_name, sorted_cols)
+    logger.info('Inserting {row_count} rows into table {table_name}'.format(row_count=row_count,
+                                                                            table_name=table_name))
+    conn = repo.get_connection()
+    cursor = conn.cursor()
+    cursor.executemany(insert_statement, tuple_list)
+    conn.commit()
+
+    if not server_was_running:
+        repo.sql_server_stop()
+
+
+def _create_table_inferred(repo: Dolt,
+                           table_name: str,
+                           data: Mapping[str, List[Any]],
+                           primary_keys: List[str]):
+    # TODO:
+    #   this sorta sucks, because it requires the Python process to own the server process, which
+    #   may no be the case if the user started it from the shell
+    assert repo.server, 'Server must be running for repo'
+
+    # generate and execute a create table statement
+    cols_to_types = {}
+    for col_name, list_of_values in data.items():
+        # Just take the first value to by the type
+        first_non_null = None
+        for val in list_of_values:
+            if val is not None:
+                first_non_null = val
+                break
+            raise ValueError('Cannot provide an empty list, types cannot be inferred')
+        cols_to_types[col_name] = _get_col_type(first_non_null, list_of_values)
+
+    query = _get_create_table_helper(table_name, cols_to_types, primary_keys)
+    conn = repo.get_connection()
+    cursor = conn.cursor()
+    cursor.execute(query)
+    conn.commit()
+    conn.close()
+
+
+def _get_col_type(sample_value: Any, values: Any):
+    if type(sample_value) == str:
+        return 'VARCHAR({})'.format(2 * max(len(val) for val in values))
+    elif type(sample_value) == int:
+        return 'INT'
+    elif type(sample_value) == float:
+        return 'F'
+    elif type(sample_value) == datetime:
+        return 'DATETIME'
+    else:
+        raise ValueError('Value of type {} is unsupported'.format(type(sample_value)))
+
+
+def _get_create_table_helper(table_name: str, cols_with_types: Mapping[str, str], pks: List[str]):
+    statement = '''
+        CREATE TABLE {table_name} (
+            {cols},
+            PRIMARY KEY ({pks})
+        )   
+    '''.format(table_name=table_name,
+               cols=','.join(['{col_name} {col_type}'.format(col_name=col_name, col_type=col_type)
+                              for col_name, col_type in cols_with_types.items()]),
+               pks=','.join(pks))
+
+    return statement
+
+
+def _get_insert_statement(table_name: str, cols: List[str]):
+    template = 'INSERT INTO {table_name} ({cols}) VALUES ({values})'
+    return template.format(table_name=table_name,
+                           cols=','.join(cols),
+                           values=','.join(['%s' for _ in range(len(cols))]))
```

Comment:
Datetime and String literal values need quotes, but numeric types do not

Reply:
The connector actually takes care of this stuff!